lib/publiccloud/provider.pm

```diff
@@ -390,10 +394,14 @@ sub terraform_apply {
     if (get_var('PUBLIC_CLOUD_SLES4SAP')) {
         $vms = $output->{cluster_nodes_name}->{value};
         $ips = $output->{cluster_nodes_public_ip}->{value};
-        my $iscsi_vms = $output->{iscsisrv_name}->{value};
-        my $iscsi_ips = $output->{iscsisrv_public_ip}->{value};
-        push @{$vms}, @{$iscsi_vms};
-        push @{$ips}, @{$iscsi_ips};
+        foreach my $others_vms (qw(drbd iscsisrv)) {
+            my $tmp_name_var = "${others_vms}_name";
+            my $tmp_ip_var   = "${others_vms}_public_ip";
+            my $tmp_vms      = $output->{${tmp_name_var}}->{value};
```

Comment:
```suggestion
            my $tmp_vms      = $output->{${tmp_name_var}}->{value} // [];
```

So you allways get a empty ARRAYREF and then a 
```
if @{$tmp_vms} 
```
should always check if it was empty...

Reply:
Or `if $tmp_vms` seems to do the job.

Reply:
**if defined $tmp_vms seems to be correct also. If all you you agree I will go with this.**